### PR TITLE
cache: remove buffer_acquire/release part 5

### PR DESCRIPTION
**Pull request by tobonex**
This is step 7 of implementation of #8006
This PR is a result of splitting https://github.com/thesofproject/sof/pull/8106 into parts.

Remove usage of buffer_acquire and buffer_release from a number of files.

next part: https://github.com/thesofproject/sof/pull/8221

### CODE REVIEW

**Review by btian1 — APPROVED** [review]

File: src/audio/module_adapter/module/waves/waves.c [hunk 1]
```diff
@@ -236,64 +234,59 @@ static int waves_effect_check(struct comp_dev *dev)
 		comp_err(dev, "waves_effect_check() source %d sink %d rate mismatch",
 			 audio_stream_get_rate(src_fmt), audio_stream_get_rate(snk_fmt));
 		ret = -EINVAL;
-		goto out;
+		return ret;
```
Comment (lyakh) [inline src/audio/module_adapter/module/waves/waves.c:237]: `return -EINVAL` - here and more below. And remove `ret`

File: src/audio/module_adapter/module/waves/waves.c [hunk 2]
```diff
@@ -322,23 +314,23 @@ static int waves_effect_init(struct processing_module *mod)
 		comp_err(dev, "waves_effect_init() sof sample format %d not supported",
 			 audio_stream_get_frm_fmt(src_fmt));
 		ret = -EINVAL;
-		goto out;
+		return ret;
```
Comment (lyakh) [inline src/audio/module_adapter/module/waves/waves.c:317]: ditto

File: src/audio/google/google_rtc_audio_processing.c [hunk 3]
```diff
@@ -610,26 +600,24 @@ static int google_rtc_audio_processing_copy(struct comp_dev *dev)
 			return ret;
 	}
 
-	buffer_c = buffer_acquire(cd->aec_reference);
-
-	ref = audio_stream_get_rptr(&buffer_c->stream);
+	ref = audio_stream_get_rptr(&cd->aec_reference->stream);
```
[the remark this answers is not in the document]
Comment (tobonex) [reply src/audio/google/google_rtc_audio_processing.c]: kept buffer_c but changed name to just buffer

**Review by lyakh — APPROVED** [review]

**Review by singalsu — APPROVED** [review]

**Review by marcinszkudlinski — APPROVED** [review]

**Comment by lgirdwood** [comment]
@tobonex @marcinszkudlinski is this next in the series ? I dont think we have any dependecies ?

**Comment by tobonex** [comment]
> @tobonex @marcinszkudlinski is this next in the series ? I dont think we have any dependecies ?
Yes, it's this one.

**Comment by lgirdwood** [comment]
@tobonex ok, lets get this in the daily test tomorrrow and if all goo we can move on to part 6.

**Pull request merged**